frontend/src/types/repository.ts

```diff
@@ -79,20 +79,83 @@ export type ExternalRepositoryInfo = {
   webUrl: string;
 };
 
-export function baseDirectoryWebUrl(repository: Repository): string {
+type WebUrlReplaceParams = {
+  DB_NAME?: string;
+  VERSION?: string;
+  TYPE?: "migrate" | "data";
+  ENV_NAME?: string;
+};
+
+export function baseDirectoryWebUrl(
+  repository: Repository,
+  params: WebUrlReplaceParams = {}
+): string {
+  let url = "";
   if (repository.vcs.type == "GITLAB_SELF_HOST") {
-    let url = `${repository.webUrl}/-/tree/${repository.branchFilter}`;
+    url = `${repository.webUrl}/-/tree/${repository.branchFilter}`;
     if (!isEmpty(repository.baseDirectory)) {
       url += `/${repository.baseDirectory}`;
     }
-    return url;
   } else if (repository.vcs.type == "GITHUB_COM") {
-    let url = `${repository.webUrl}/tree/${repository.branchFilter}`;
+    url = `${repository.webUrl}/tree/${repository.branchFilter}`;
     if (!isEmpty(repository.baseDirectory)) {
       url += `/${repository.baseDirectory}`;
     }
+  }
+  if (url) {
+    // BYT-1344. Replace the patterns in the filePathTemplate if possible.
```

Comment:
Maybe we don't need it in code.
```suggestion
    // Replace the patterns in the filePathTemplate if possible.
```